sdk/confidentialledger/azure-security-confidentialledger/src/test/java/com/azure/security/confidentialledger/ConfidentialLedgerClientTestBase.java

```diff
@@ -1,104 +1,92 @@
-            policies.add(new BearerTokenAuthenticationPolicy(credential, ConfidentialLedgerClientBuilder.DEFAULT_SCOPES));
-        }
-
-        policies.add(new RetryPolicy());
-
-        HttpPolicyProviders.addAfterRetryPolicies(policies);
-        policies.add(new HttpLoggingPolicy(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)));
-
-        if (interceptorManager.isPlaybackMode()) {
-            httpClient = interceptorManager.getPlaybackClient();
-        } else {
-            try {
-                byte[] pemKey = Files.readAllBytes(Paths.get(
-                    Configuration.getGlobalConfiguration().get("CONFIDENTIALLEDGER_KEY_PATH")));
-                reactor.netty.http.client.HttpClient reactorClient = reactor.netty.http.client.HttpClient.create()
-                    .secure(sslContextSpec -> sslContextSpec.sslContext(SslContextBuilder.forClient()
-                        .trustManager(new ByteArrayInputStream(pemKey))));
-                httpClient = new NettyAsyncHttpClientBuilder(reactorClient).wiretap(true).build();
-            } catch (IOException e) {
-                throw new RuntimeException(e);
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+
+class ConfidentialLedgerClientTestBase extends TestBase {
+    protected ConfidentialLedgerClient confidentialLedgerClient;
+
+    @Override
+    protected void beforeTest() {
+        try {
+            ConfidentialLedgerIdentityClientBuilder confidentialLedgerIdentityClientbuilder = new ConfidentialLedgerIdentityClientBuilder()
+                .identityServiceUri(
+                        Configuration.getGlobalConfiguration().get("IDENTITYSERVICEURI", "identityserviceuri"))
+                .httpClient(HttpClient.createDefault())
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+            if (getTestMode() == TestMode.PLAYBACK) {
+                confidentialLedgerIdentityClientbuilder
+                        .httpClient(interceptorManager.getPlaybackClient())
+                        .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
+            } else if (getTestMode() == TestMode.RECORD) {
+                confidentialLedgerIdentityClientbuilder
+                        .addPolicy(interceptorManager.getRecordPolicy())
+                        .credential(new DefaultAzureCredentialBuilder().build());
+            } else if (getTestMode() == TestMode.LIVE) {
+                confidentialLedgerIdentityClientbuilder.credential(new DefaultAzureCredentialBuilder().build());
+            }
+            ConfidentialLedgerIdentityClient confidentialLedgerIdentityClient = confidentialLedgerIdentityClientbuilder
+                    .buildClient();
+            String ledgerId = Configuration.getGlobalConfiguration().get("LEDGERURI")
+                    .replaceAll("\\w+://", "")
+                    .replaceAll("\\..*", "");
```

Comment:
please run the test (without "RECORD" locally).

this line and some more is wrong. In test env on CI, there is no such environment variable defined. You had to give them a default value.